src/main/java/com/github/slavaz/maven/plugin/postgresql/embedded/goals/AbstractGoalMojo.java

```diff
@@ -1,25 +1,91 @@
 package com.github.slavaz.maven.plugin.postgresql.embedded.goals;
 
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.IterableUtils;
+import org.apache.commons.collections4.Predicate;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
 
 /**
  * Created by slavaz on 13/02/17.
  */
 public abstract class AbstractGoalMojo extends AbstractMojo {
 
-    @Parameter(defaultValue = "false")
-    private boolean skipGoal;
+	protected static final String SP_HTTP_PROXY_HOST = "http.proxyHost";
+
+	protected static final String SP_HTTP_PROXY_PORT = "http.proxyPort";
+
+	protected static final String SP_HTTP_NO_PROXY = "http.nonProxyHosts";
+
+	protected static final String PROTOCOL_HTTP = "http";
+
+	@Parameter(defaultValue = "false")
+	private boolean skipGoal;
+
+	@Parameter(defaultValue = "${settings}", readonly = true)
+	protected Settings settings;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skipGoal) {
+			getLog().debug("Skipped.");
+		} else {
```

Comment:
Would be great to call  handleProxyConfigurjation() here and to keep doExecute() as abstract